src/commons/navigationBar/subcomponents/SicpNavigationBar.tsx

```diff
@@ -63,15 +98,412 @@ const SicpNavigationBar: React.FC = () => {
     usePortal: false
   };
 
+  const searchDrawerProps = {
+    onClose: handleCloseSearch,
+    autoFocus: true,
+    canEscapeKeyClose: true,
+    canOutsideClickClose: true,
+    enforceFocus: true,
+    hasBackdrop: true,
+    isOpen: isSearchOpen,
+    position: Position.LEFT,
+    usePortal: false
+  };
+
+  function queryIndexTrie(query: string) {
+    let node = indexTrie;
+    for (let i = 0; i < query.length; i++) {
+      const char = query[i];
+      if (node[char]) {
+        node = node[char];
+      } else {
+        return [];
+      }
+    }
+    if (node['value']) {
+      return node['value'];
+    }
+    return [];
+  }
+
+  function queryTextTrie(query: string) {
+    let node = textTrie;
+    for (let i = 0; i < query.length; i++) {
+      const char = query[i];
+      if (node[char]) {
+        node = node[char];
+      } else {
+        return [];
+      }
+    }
+    if (node['value']) {
+      return node['value'];
+    }
+    return [];
+  }
```

Comment:
There's a lot of duplicated code here. There should just be one function, with the trie being queried passed in as an argument instead of relying on closure.

Reply:
I've fixed this in #2471. Do you mind taking a look?